Centralize definition of time zero

cmcantalupo opened the pull request.

cmcantalupo committed:
Add a function that returns a universal zero time

Change-Id: I990fb6bf0b933b4233b11135e9e44de7be59fb27
Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Swich all zero time calculations to the TimeZero supported value

Change-Id: Icaf3a0076e500667cd54f01c4b6ecd2dc3299d3c
Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Remove unused PlatformIO instance

Change-Id: Ie5b78cd19aac3570ad7d0b7141ab61b1a283f24a
Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

bgeltz reviewed (changes requested):
Something looks fundamentally broken in the integration tests.

Code LGTM though.

cmcantalupo committed:
Fixup ApplicationSampler test

Change-Id: Ic6b258528c1d69811f87b756f955c95c90a95cf4
Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Fix the RuntimeRegulator use of time_zero()

- The m_time_zero member of the RuntimeRegulator seems to serve
  the sole purpose that when m_rank_log[].enter_time is set to
  that value, then it is inferred tha the last message about
  the region was an exit call.  Since this has nothing to do with
  time zero, this patch renames the variable and sets it to {{0,0}}.

Change-Id: Id37e010c60f9d6cd4954c9d280a892cd7c3a62ab
Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

bgeltz reviewed (dismissed):
The exit cookie thing is gross.  I'm wondering if it would be better to just write out {0,0} in all the spots?  Anyway, approved.
